### PR TITLE
GitHub Action for updating `messages.pot`

### PR DESCRIPTION
**Pull request by meowmeowmeowcat**
This Github Action runs every day automatically for updating `messages.pot`.

### CODE REVIEW

**Comment by bhrutledge** [comment]
I didn't review https://github.com/pypa/packaging.python.org/pull/935, but based on https://github.com/pypa/packaging.python.org/pull/937/commits/ede3fed7c2f78cc756ff5b5818a62f8f7a1336c2, it seems like there's still potential for `messages.pot` to be updated during local development, and accidentally included in PR's.

I'm really not familiar with translations; is it common practice to include `*.pot` files in the repo?

**Comment by meowmeowmeowcat** [comment]
@bhrutledge We have to update the `messages.pot` locally and commit the change after we changed the content of html pages in Warehouse. 
Here's example: PR [#9690](https://github.com/pypa/warehouse/pull/9690/files#diff-7cc1dd5466d4caf26bedf81acb95231b510147da1a3ed4ff5a0ae3f8fa3eeb3a) in Warehouse. 

However, our main purpose is maintaining the docs, so our docs will be changed constantly. 
This makes me to think that if we should open a new repo for translations of this docs so folks won't accidentally included `messages.pot` in PR.

**Comment by meowmeowmeowcat** [comment]
> I'm really not familiar with translations; is it common practice to include *.pot files in the repo?

yes, it's really common to include *.pot files in the repo, just like Warehouse.

File: .github/workflows/translation.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: Translation
+
+on:
+  schedule:
+    - cron: '00 00,00 * * *'
```
Comment (webknjaz) [inline .github/workflows/translation.yml:5]: This cron spec doesn't look right.

File: .github/workflows/translation.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        run: python -m pip install --upgrade nox virtualenv
+        
+      - name: Create local changes
+        run: python -m nox -s translation
+        
+      - name: Commit files
+        run: |
+          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git commit -m "Update messages.pot" -a
```
Comment (webknjaz) [inline .github/workflows/translation.yml:29]: I think it'd be useful to include the current commit hash in the message (the short one would do).

Comment (webknjaz) [reply .github/workflows/translation.yml:29]: Also, I'd not use `-a` but just point to the pot file.

File: .github/workflows/translation.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+      - name: Install Dependencies
+        run: python -m pip install --upgrade nox virtualenv
+        
+      - name: Create local changes
+        run: python -m nox -s translation
+        
+      - name: Commit files
+        run: |
+          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
```
Comment (webknjaz) [inline .github/workflows/translation.yml:27]: Drop that `number+` thing

File: .github/workflows/translation.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+          git config --local user.name "github-actions[bot]"
+          git commit -m "Update messages.pot" -a
+        
+      - name: Push changes
+        uses: ad-m/github-push-action@master
+        with:
+         github_token: ${{ secrets.ACCESS_TOKEN }}
+         branch: ${{ github.ref }}
```
Comment (webknjaz) [inline .github/workflows/translation.yml:35]: This is unnecessary, just do `git push --atomic`

Comment (webknjaz) [reply .github/workflows/translation.yml:35]: (specify what to push tho)

File: .github/workflows/translation.yml (hunk 1 above)
Comment (meowmeowmeowcat) [reply .github/workflows/translation.yml:5]: Also, I would prefer less commits to the repo, so maybe I should set it to update automatically on every Sunday?

Comment (meowmeowmeowcat) [reply .github/workflows/translation.yml:5]: I think this is right.

Image

**Comment by meowmeowmeowcat** [comment]
Moved this PR to #945

**Pull request closed without merging**